### PR TITLE
Wait for BookKeeper Cluster to start

### PR DESCRIPTION
**Pull request by eolivelli**
- Wait for bookie to be up and running while boostrapping a Tablespace (server.bookkeeper.wait.cluser.ready.timeout)
- Upgrade to ZooKeeper 3.6.2
- Clean up log level
- Use default expectedreplicacount even for "herd" tablespace

With this change is will be easy to start an HerdDB cluster with default expectedreplicacount greater than one, because the first server will have time to wait for the other servers to start the internal bookie

set server.bookkeeper.wait.cluser.ready.timeout=0 in order to rollback to previous behaviour

### CODE REVIEW

File: herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLogManager.java [hunk 1]
```diff
@@ -62,10 +62,13 @@
     private long ledgersRetentionPeriod = 1000 * 60 * 60 * 24;
     private long maxLedgerSizeBytes = 100 * 1024 * 1024 * 1024;
     private long maxIdleTime = 0;
+    private long bookkeeperClusterReadyWaitTime = 60_000;
+
     private ConcurrentHashMap<String, BookkeeperCommitLog> activeLogs = new ConcurrentHashMap<>();
 
     public BookkeeperCommitLogManager(ZookeeperMetadataStorageManager metadataStorageManager, ServerConfiguration serverConfiguration, StatsLogger statsLogger) {
         this.statsLogger = statsLogger;
+        this.bookkeeperClusterReadyWaitTime = serverConfiguration.getInt(ServerConfiguration.PROPERTY_BOOKKEEPER_WAIT_CLUSTER_READY_TIMEOUT, ServerConfiguration.PROPERTY_BOOKKEEPER_WAIT_CLUSTER_READY_TIMEOUT_DEFAULT);
```
Comment (nicoloboschi) [inline herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLogManager.java:71]: getLong? or convert bookkeeperClusterReadyWaitTime to int

File: herddb-core/src/main/java/herddb/server/ServerConfiguration.java [hunk 2]
```diff
@@ -148,6 +148,9 @@
     public static final String PROPERTY_BOOKKEEPER_LEDGERS_RETENTION_PERIOD = "server.bookkeeper.ledgers.retention.period";
     public static final long PROPERTY_BOOKKEEPER_LEDGERS_RETENTION_PERIOD_DEFAULT = 1000L * 60 * 60 * 24 * 2;
 
+    public static final String PROPERTY_BOOKKEEPER_WAIT_CLUSTER_READY_TIMEOUT = "server.bookkeeper.wait.cluser.ready.timeout";
+    public static final int PROPERTY_BOOKKEEPER_WAIT_CLUSTER_READY_TIMEOUT_DEFAULT = 6000 * 2; // 2 minutes
```
Comment (nicoloboschi) [inline herddb-core/src/main/java/herddb/server/ServerConfiguration.java:152]: did you mean 60_000 * 2? is it in ms right?

File: herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java [hunk 3]
```diff
@@ -166,6 +159,41 @@ private CommitFileWriter() throws LogNotAvailableException {
             }
         }
 
+        private WriteHandle makeNewWriteHandle(int actualEnsembleSize, int actualWriteQuorumSize, int actualAckQuorumSize,
+                                               Map<String, byte[]> metadata) throws BKException, LogNotAvailableException {
+            BKNotEnoughBookiesException lastError = null;
+            long maxTime = System.currentTimeMillis() + parent.getBookkeeperClusterReadyWaitTime();
```
Comment (nicoloboschi) [inline herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java:165]: final?

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
Good work, I left some comments
I think default timeout value is incorrect

File: herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLogManager.java (hunk 1 above)
Comment (eolivelli) [reply herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLogManager.java:71]: good catch

File: herddb-core/src/main/java/herddb/server/ServerConfiguration.java (hunk 2 above)
Comment (eolivelli) [reply herddb-core/src/main/java/herddb/server/ServerConfiguration.java:152]: good catch

**Comment by eolivelli** [comment]
@nicoloboschi I have addressed all of your comments. PTAL

**Review by nicoloboschi — APPROVED** [review]

**Review by diegosalvi — APPROVED** [review]

**Pull request merged**